task/thermal/src/bsp/cosmo_a.rs

```diff
@@ -0,0 +1,370 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! BSP for the Cosmo rev A hardware
+
+use crate::{
+    control::{
+        ChannelType, ControllerInitError, Device, FanControl, Fans,
+        InputChannel, Max31790State, PidConfig, TemperatureSensor,
+    },
+    i2c_config::{devices, sensors},
+};
+pub use drv_cpu_seq_api::SeqError;
+use drv_cpu_seq_api::{PowerState, Sequencer, StateChangeReason};
+use task_sensor_api::SensorId;
+use task_thermal_api::ThermalProperties;
+use userlib::{task_slot, units::Celsius, TaskId, UnwrapLite};
+
+task_slot!(SEQ, cosmo_seq);
+
+// We monitor the TMP117 air temperature sensors, but don't use them as part of
+// the control loop.
+const NUM_TEMPERATURE_SENSORS: usize = sensors::NUM_TMP117_TEMPERATURE_SENSORS;
+
+const NUM_NVME_BMC_TEMPERATURE_SENSORS: usize =
+    sensors::NUM_NVME_BMC_TEMPERATURE_SENSORS;
+
+// The control loop is driven by CPU, NIC, and BMC temperatures
+// XXX we should also monitor DIMM temperatures here
```

Comment:
@nathanaelhuffman / @Aaron-Hartwig I think we need integration with the main FGPA here, because we'll be reading DIMM temperatures through the I3C proxy.

Reply:
This is true though may be a post-bringup activity. We've been having a number of issues with the ruby+grapefruit signal integrity to properly prototype this and while I'm still working to improve the test fixture over the next couple of days, we're running out of time before lab pack-up must commence.